rlp/rlp.go

```diff
@@ -0,0 +1,43 @@
+package rlp
+
+import (
+	"encoding/binary"
+	"math"
+)
+
+type Item struct {
+	D []byte
+	L []*Item
+}
+
+func Encode(input *Item) []byte {
```

Comment:
Do you have to do a `if input == nil` check here?

Reply:
Perhaps. Although I don't know what you should expect to have happen if you call `Encode(nil)` Perhaps a panic is fine behavior.

Reply:
the latest code does error checking on `input`